src/test/java/io/confluent/connect/jdbc/source/integration/MSSQLDateTimeIT.java

```diff
@@ -0,0 +1,298 @@
+        // Setup up props for the source connector
+        props = configProperties();
+        // set to TimeStamp + Incrementing mode
+        props.put(MODE_CONFIG, MODE_TIMESTAMP_INCREMENTING);
+        props.put(INCREMENTING_COLUMN_NAME_CONFIG, "incrementing_col");
+
+
+        String sql = "CREATE TABLE " + MSSQL_Table
+                + " (start_time DATETIME not NULL, incrementing_col int not NULL, "
+                + "record VARCHAR(255))";
+        PreparedStatement createStmt = connection.prepareStatement(sql);
+        executeSQL(createStmt);
+
+
+        // Create topic in Kafka
+        KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+
+        // Configure source connector
+        configureAndWaitForConnector();
+
+        Timestamp t = Timestamp.from(
+                ZonedDateTime.of(2016, 12, 8, 19, 34, 56, 0, ZoneId.of("UTC")).toInstant()
+        );
+
+        sql = "INSERT INTO " + MSSQL_Table + " (start_time, incrementing_col, record) "
+                + "values (?, ?, ?)";
+        PreparedStatement insertStmt = connection.prepareStatement(sql);
+        insertStmt.setTimestamp(1, t);
+        insertStmt.setInt(2, 1);
+        insertStmt.setString(3, "example record value");
+        executeSQL(insertStmt);
+
+
+        connect.assertions().assertConnectorIsRunningAndTasksHaveFailed(
+                CONNECTOR_NAME,
+                Math.min(KAFKA_TOPICS.size(), TASKS_MAX),
+                "failed to verify that tasks have failed"
+        );
+    }
+
+
+    /**
+     * Verify that Datetime as a non timestamp column does not kill task
+     *
+     * Also Verify that after a number of poll intervals the number of records in topic
+     * still equal to the number of records written to the MSSQL Server instance.
+     * This confirms that DateTime2 works as expected.
+     *
+     */
+    @Test
+    public void verifyDateTimeAllowedAsNonTimeStamp() throws Exception {
+        props = configProperties();
+        props.put(MODE_CONFIG, MODE_TIMESTAMP);
+
+
+        String sql = "CREATE TABLE " + MSSQL_Table
+                + " (start_time DATETIME2 not NULL, record DATETIME)";
+        PreparedStatement createStmt = connection.prepareStatement(sql);
+        executeSQL(createStmt);
+
+        // Create topic in Kafka
+        KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+
+        // Configure source connector
+        configureAndWaitForConnector();
+
+        Timestamp t = Timestamp.from(
+                ZonedDateTime.of(2016, 12, 8, 19, 34, 56, 0, ZoneId.of("UTC")).toInstant()
+        );
+
+        sql = "INSERT INTO " + MSSQL_Table + " (start_time, record) "
+                + "values (?, ?)";
+        PreparedStatement insertStmt = connection.prepareStatement(sql);
+        insertStmt.setTimestamp(1, t);
+        insertStmt.setTimestamp(2, t);
+        executeSQL(insertStmt);
+
+
+        Thread.sleep(Duration.ofSeconds(30).toMillis());
```

Comment:
Are you waiting for something specific here? Can we do a wait-and-probe type strategy, instead of just 30 seconds?

Reply:
We're just verifying that when `datetime2` is used the number of rows written to MSSQL is the number rows read from it. The way to do this is just wait for long enough that the `TableQuerier` would have polled the MSSQL Table more than once checking for new records. There's no real wait-and-probe strategy cause in the test `pass` case  no matter how long we wait we should still have the same number of records read! I'll leave a comment